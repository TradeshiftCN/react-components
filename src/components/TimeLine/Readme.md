```jsx
const timelines = [
  {
    id: 'uuid1',
    time: '2019-07-01 15:20',
    status: '处理成功',
    name: 'puff'
  },
  {
    id: 'uuid2',
    time: '2019-06-30 13:54',
    status: '创建成功',
    name: 'puff'
  }
];
<TimeLine timelines={timelines} activeIndex={0} />;
```
