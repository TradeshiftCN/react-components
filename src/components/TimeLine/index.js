import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import './index.less';

class TimeLine extends React.Component {
  render() {
    const { timelines } = this.props;
    return (
      <div>
        {timelines.map((timeline, index) => (
          <div
            className={`timeline ${this.props.activeIndex === index &&
              'active'}`}
            key={timeline.time}
          >
            <i className="circle" />
            <div className="content">
              <div className="time">
                {moment(timeline.time).format('YYYY-MM-DD HH:mm')}
              </div>
              <div className="status">{timeline.status}</div>
              <div className="number">{timeline.name}</div>
            </div>
          </div>
        ))}
      </div>
    );
  }
}

TimeLine.propTypes = {
  // TimeLine配置数组
  timelines: PropTypes.arrayOf(
    PropTypes.shape({
      time: PropTypes.string,
      status: PropTypes.string,
      name: PropTypes.string
    })
  ),
  // 当前节点下标
  activeIndex: PropTypes.number
};

export default TimeLine;
