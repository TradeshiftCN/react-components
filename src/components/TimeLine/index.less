@import '../../style/variables/index';

.timeline {
	padding-left: @ts-unit;
	position: relative;
	width: 100%;
	&::before {
		content: '';
		position: absolute;
		top: 0;
		bottom: 0;
		left: @ts-unit-quarter;
		width: 2px;
		background-color: @ts-color-gray;
	}
	.content {
		color: @ts-color-slate-lighter;
		border-bottom: dashed 1px @ts-color-gray;
		padding-bottom: @ts-unit-half;
		padding-left: @ts-unit-quarter;
		opacity: 0.6;
		.time {
			color: @ts-color-slate;
			font-size: @ts-fontsize;
			line-height: @ts-unit;
			margin-bottom: @ts-unit-quarter;
			margin-top: @ts-unit-quarter;
			font-weight: @ts-fontweight-semibold;
		}

		.status,
		.number {
			font-size: @ts-fontsize-mini;
		}
	}

	.circle {
		position: absolute;
		top: -4px;
		left: -4px;
		width: @ts-unit;
		height: @ts-unit;
		background-color: @ts-color-gray;
		border-radius: @ts-unit-half;
	}

	&.active {
		.content {
			opacity: 1;
			.time {
				margin-top: 0;
			}
		}
		.circle {
			background-color: @ts-color-blue;
			top: 0;
			&::before {
				content: '';
				position: absolute;
				left: 1px;
				top: 1px;
				width: 18px;
				height: 18px;
				border-radius: @ts-unit-half;
				border: solid 1px @ts-color-white;
			}
		}
	}
}
