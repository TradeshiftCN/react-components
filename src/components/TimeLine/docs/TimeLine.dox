/**
 * @title 基本功能
 * @renderModel left
 */
import TimeLine from '@tradeshift/react-components/lib/TimeLine'

const TimeLineBasic = () => {
  const timelines = [
      {
          time: '2019-07-01 15:20',
          status: '处理成功',
          name: 'puff'
      },
      {
          time: '2019-06-30 13:54',
          status: '创建成功',
          name: 'puff'
      }
  ];
  return (
     <TimeLine timelines={timelines} activeIndex={0} />
  )
}

@component TimeLine
