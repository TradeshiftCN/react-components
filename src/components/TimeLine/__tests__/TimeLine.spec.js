import React from 'react';
import { shallow } from 'enzyme';
import TimeLine from '../index';

describe('TimeLine', () => {
  it('should render correctly', () => {
    const props = {
      timelines: [
        {
          time: '2019-07-01 15:20',
          status: '处理成功',
          name: 'puff'
        },
        {
          time: '2019-06-30 13:54',
          status: '创建成功',
          name: 'puff'
        }
      ],
      activeIndex: 0
    };
    const wrapper = shallow(<TimeLine {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
