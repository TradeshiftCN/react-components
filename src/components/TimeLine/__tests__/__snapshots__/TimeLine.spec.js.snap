// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeLine should render correctly 1`] = `
<div>
  <div
    className="timeline active"
    key="2019-07-01 15:20"
  >
    <i
      className="circle"
    />
    <div
      className="content"
    >
      <div
        className="time"
      >
        2019-07-01 15:20
      </div>
      <div
        className="status"
      >
        处理成功
      </div>
      <div
        className="number"
      >
        puff
      </div>
    </div>
  </div>
  <div
    className="timeline false"
    key="2019-06-30 13:54"
  >
    <i
      className="circle"
    />
    <div
      className="content"
    >
      <div
        className="time"
      >
        2019-06-30 13:54
      </div>
      <div
        className="status"
      >
        创建成功
      </div>
      <div
        className="number"
      >
        puff
      </div>
    </div>
  </div>
</div>
`;
